### ISSUE TITLE
Support for Android Studio line wrapping

### ISSUE THREAD

**Issue by vibhorgoswami** [diffplug/spotless#817]
While using spotless for Android, we run in to an issue where Android Studio line wrapping is overridden with spotless. There is no API exposed for changing this so that we can synchronize Android studio line wrapping with Spotless line wrapping.
The formatting of java code is being set to `googleJavaFormat(1.8).aosp()` and having a customizable line wrapping would help with unwanted git changes issues.
`java {
        target '**/*.java'
        targetExclude '**/generated/**'
        googleJavaFormat('1.8').aosp()
        removeUnusedImports()
        trimTrailingWhitespace()
        indentWithSpaces()
        endWithNewline()
    }`

**Comment by jbduncan** [comment]
It's known that the formatting of google-java-format differs to IntelliJ's (and thus Android Studio's). It's in fact a consequence of the google-java-format team's goal to make a formatter that was more flexible for them than IntelliJ's formatter was at the time.

I've just seen that @nedtwigg labelled this as "won't fix" and I can see why...

**Comment by jbduncan** [comment]
Never mind, I was too quick re. the "won't fix" comment! Apologies!

**Comment by jbduncan** [comment]
I think people have managed to "hack" the internals of google-java-format in the past to allow the line wrapping length to be adjusted. See [talios/googleformatter-maven-plugin](https://github.com/talios/googleformatter-maven-plugin) for a Maven plugin example of how this might be done.

But as always, this is prone to breakage when a new version of google-java-format is released.

**Comment by nedtwigg** [comment]
I agree with everything @jbduncan said, and I think this is mostly a wontfix. But! It is somewhat frequent for users to complain about line-wrapping behavior. One solution might be a`dontChangeLineWrap()` step which could work with any formatter, not just `googleJavaFormat()`. It would work like so:

- store the lines of the file at very beginning
- run all other steps
- look at the final result, and diff it with the original
- for any diff which is just a linewrap, go back to the original wrap

This is tricky to build, and it's not going to make my todo list. But if someone wants to do it, these parts are already available:

- `PipeStepPair` can do "store at beginning, then do stuff again after all the other steps" (see #691)
- JGit can do fast line diffs, figuring out which are pure line-wrap and undoing just those deltas is tricky
